bundler/lib/bundler/definition.rb

```diff
@@ -89,6 +89,7 @@ def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, opti
         @locked_gems = LockfileParser.new(@lockfile_contents)
         @locked_platforms = @locked_gems.platforms
         @platforms = @locked_platforms.dup
+        platforms.each {|p| add_platform(p) }
```

Comment:
does it work to call `platforms.each {|p| add_platform(p) }` unconditionally, at line 118?

Reply:
Probably. This was a premature optimization when we know that `@locked_platforms` is empty when there's no lockfile.